flow/layers/layer.cc

```diff
@@ -77,7 +76,6 @@ Layer::AutoSaveLayer::AutoSaveLayer(const PaintContext& paint_context,
       canvas_(save_mode == SaveMode::kInternalNodesCanvas
                   ? *(paint_context.internal_nodes_canvas)
                   : *(paint_context.leaf_nodes_canvas)) {
-  TRACE_EVENT0("flutter", "Canvas::saveLayer");
```

Comment:
removing the saveLayer events will break DevTools. We have logic to count these events and provide a hint to users when we detect this expensive operation in their janky frame.

Reply:
So I can certainly leave these in the tree, but I don't think relying on this is correct. You are only going to see saveLayers that are composited by flutter, you will miss all of the saveLayers that are performed directly on cavnas. That could easily be the majority of cases

Reply:
done

Reply:
Okay gotcha. This was the event I was told to check for at the time these were added - @dnfield what is the best way to catch all the save layer events from flutter and from the canvas?

Reply:
Its not going to be possible without instrumenting the display list. that said, given that these trace events are slow we should find a different way to record this information.

Reply:
saveLayer opeations on canvas should not happen very often. Finding a different way to instrument this is fine, but we should make sure to not break devtools in the mean time.

Reply:
Thanks to this discussion I realized that I had not supported these events in the new LayerStateStack stuff.

https://github.com/flutter/engine/pull/36458/commits/7632996a3e0fa57db4b604f99a8939b7e53b9dc6